src/core/services/theming/theming.js

```diff
@@ -528,7 +528,7 @@ function generateAllThemes($injector) {
   if (generateOnDemand) return;
 
   angular.forEach(THEMES, function(theme) {
-    if (!GENERATED[theme.name]) {
+    if (!GENERATED[theme.name] && !(theme.name !== $mdTheming.defaultTheme() && theme.name === 'default')) {
```

Comment:
Let's go through a few cases.

Case 1
-----
We have the following mocked theme array
```js
THEMES = ['default', ' blue']
```
And my default theme is set to `blue`

So the iteration will check:
```js
// Replicates the iteration - case for `default`
var themeName = THEMES[0]; // `default`
// Negate the check, so if both conditions are true, 
// then the theme won't be inserted into the DOM
if (!(
  themeName !== 'blue' // true -> continue with the checks `AND`
  && themeName === 'default' // true -> both conditions are true
)
```
So in this case, only the theme `blue` will inserted into the DOM - <kbd>Valid</kbd>

Case 2
---
We have the following mocked theme array
```js
THEMES = ['default', ' blue']
```
And my default theme is set to `default` (default behavior)

So the iteration will check:
```js
// Replicates the iteration - case for `default`
var themeName = THEMES[0]; // `default`
// Negate the check, so if both conditions are true, 
// then the theme won't be inserted into the DOM
if (!(
  themeName !== 'default' // false -> cancel check and and return false
  && themeName === 'default' // Not reachable
)
// The check will pass, because of the negated expression 
// !(false) ==> true
```
So in this case, both themes will be inserted to the DOM